templates/fine-tune-llm_v2/cookbooks/lora_vs_full_parameter/README.md

```diff
@@ -0,0 +1,98 @@
+# Fine-Tuning with LoRA vs full-parameter
+
+**⏱️ Time to complete**: 45 minutes (for the LoRA fine-tuning)
+
+This guide assumes that you have familiarized yourself with the [main fine-tuning guide](../../README.md) of this template.
+In this cookbook, we explain the nuances of fine-tuning with [LoRA (Low-Rank Adaptation)](https://arxiv.org/abs/2106.09685) versus full-parameter fine-tuning.
+
+## Quick theoretical comparison
+
+Full-parameter fine-tuning takes the LLM "as is" and trains it on the given dataset. In principle, this is regular supervised training like in the pretraining stage of the LLM. You can expect full-parameter fine-tuning to result in slightly higher model quality.
+
+[LoRA (Low-Rank Adaptation)](https://arxiv.org/abs/2106.09685) is a fine-tuning technique that freezes all the weights of your LLM and adds a few parameters to it that get fine-tuned instead. These additional parameters make up a LoRA checkpoint. There are two important things to take away from this:
+1. Since all the original weights are frozen, they don't have to be optimized and therefore don't take up as many resources during fine-tuning. In practice, you can fine-tune on a smaller cluster.
+2. Since the checkpoint only consists of the few additional parameters, it is very small. If we load the original model into memory, we can swap out the fine-tuned weights quickly. Therefore, it makes for an efficient scheme for serving many fine-tuned models alongside each other.
+
+You can find a more in-depth analysis of this topic [here](https://www.anyscale.com/blog/fine-tuning-llms-lora-or-full-parameter-an-in-depth-analysis-with-llama-2).
```

Comment:
I would add some more recent research here as well:
- https://arxiv.org/abs/2405.09673

Reply:
Yeah that's the paper I had in mind the other day, only that it was not from HF 😄